requirements/dev

```diff
@@ -0,0 +1,25 @@
+# Please make sure to pin all dependency versions, in order to avoid unwanted
+# functional and integration breaks caused by external code updates.
+#
+# These are dev dependencies, so pin them.
+
+# For automatic style formatting.
+autopep8 == 1.5.7
+
+# For style & code checking.
+flake8 == 3.9.1
+flake8-bugbear == 21.4.3
+flake8-print == 4.0.0
+
+# For PyTest test coverage integration.
+pytest-cov == 2.11.1
+
+# For optional duck & static type checking.
+mypy == 0.812
+
+# For testing: parameters, variables, etc.
+openfisca-country-template == 3.12.5
+openfisca-extension-template == 1.3.6
```

Comment:
Pinning those seems risky to me, as we are likely to lose the benefits from #1020.

Reply:
There's a bit of a tricky decision to be made here, the context:

- Core & the templates are circular dependencies
- That prevents us of using the latest pip version, without the legacy resolver
- In order to fix that, several scenarios, notably two:
  - Relax versions somewhere, and pin them somewhere else —as proposed here
  - Extract the dependencies, with the trade-off of losing integration tests, as discussed already in another issue.

I think this requires a proper issue, exclusively to fix the `pip` problem.

As proposed [here](https://github.com/openfisca/openfisca-core/pull/1020#issue-629112079):

> Force to phase breaking changes so to adapt the packages depending on OpenFisca-Core

Reply:
I proposed changes unpinning them, please let me know what do you think.